Add explicit type declarations to class constants (PHP 8.3+)

lacatoire opened the pull request:
### Summary
This PR introduces explicit type declarations for class constants, leveraging the feature introduced in PHP 8.3.

### Motivation
- Enhances type safety and static analysis.
- Makes code intent clearer.
- Prepares PhD for PHP 8.4 compatibility and future strictness.

### Scope
- Added types such as `int`, `string`, `bool`, or `array` to class constants.
- Example:
    Before:
        public const FORMAT_HTML = 1;
        public const NAME = 'PhD';
    After:
        public const int FORMAT_HTML = 1;
        public const string NAME = 'PhD';

### Impact
- ✅ No runtime behavior change.
- ✅ Fully backward compatible with PHP 8.3+.
- ⚙️ Code clarity and safety improvement only.

### References
- [PHP 8.3: Typed class constants RFC](https://wiki.php.net/rfc/typed_class_constants)
- [PHP manual: Class constants](https://www.php.net/manual/en/language.oop5.constants.php)

Commit:
refactor: add explicit type declarations to class constants (PHP 8.3+)

Girgias reviewed (changes requested):
Can you update the composer.json file to require 8.3? I checked with Derick and the doc machine runs 8.3 so it should be fine.

lacatoire committed:
Merge branch 'php:master' into refactor/add-typed-class-constants

lacatoire committed:
Merge branch 'master' into refactor/add-typed-class-constants

lacatoire committed:
Merge remote-tracking branch 'origin/refactor/add-typed-class-constants' into refactor/add-typed-class-constants

lacatoire committed:
setting php 8.3

lacatoire committed:
Fix <function> with <replaceable> not rendering correctly

When <function> contains <replaceable>, skip link/parentheses
processing in format_function_text() and output plain text instead.
Uses the role stack pattern already established by format_constant().

Fixes php/phd#171

lacatoire committed:
rm 8.1 & 8.2 in CI

lacatoire committed:
Merge branch 'master' into refactor/add-typed-class-constants

kocsismate commented:
@lacatoire At last, I had to revert these changes because it turned out that Gina got incorrect information, and the servers still run on PHP 8.2, and that's why the recent docs changes didn't synchronize to php.net.

cweiske commented:
The [Dockerfile in doc-en](https://github.com/php/doc-en/blob/master/.docker/Dockerfile) also still uses php 8.2.
